spec/requests/chat_controller_spec.rb

```diff
@@ -208,6 +208,49 @@
     end
   end
 
+  describe "#rebake" do
+    fab!(:chat_channel) { Fabricate(:chat_channel) }
+    fab!(:chat_message) { Fabricate(:chat_message, chat_channel: chat_channel, user: user) }
+
+    context "staff" do
+      it "rebakes the post" do
+        sign_in(Fabricate(:admin))
+
+        expect do
+          put "/chat/#{chat_channel.id}/#{chat_message.id}/rebake.json"
+        end.to change { Jobs::ProcessChatMessage.jobs.size }.by(1)
+        expect(response.status).to eq(200)
+
+        job = Jobs::ProcessChatMessage.jobs.first
+        job_args = job["args"].first
+        expect(job_args["chat_message_id"]).to eq(chat_message.id)
+        expect(job_args["is_dirty"]).to be_blank
+        ChatPublisher.expects(:publish_processed!).never
+        job["class"].constantize.new.perform(*job["args"])
```

Comment:
This assertion is a little odd. Why do we need to manually execute a job in a request spec? Perhaps you need to enable `Jobs.run_immediately!`? Also, we should try and avoid mocking as much as possible. Instead you can use `MessageBus.track_publish` to check that no message was published here.

Reply:
Yep I started with run_immediately but it prevented me to catch jobs size. ON second thought though, I just think this logic should be in the ProcessChatMessage spec, which is what I did. Concerning using `track_publish`, I'm also mixed on this, because we are testing that `ChatPublisher. publish_processed!` will be called, not that message bus will trigger a message, `publish_processed` implementation could change in the future and it's not the responsibility of the job to test the underlying implementation. I'm opened to a different refactor if you have a suggestion.

Reply:
>  because we are testing that ChatPublisher. publish_processed! will be called, not that message bus will trigger a message, publish_processed implementation could change in the future and it's not the responsibility of the job to test the underlying implementation.

I disagree here because by testing that `ChatPublisher.publish_processed!` will be called, you're testing for the underlying implementation of the job. Asserting that a message_bus message is published is testing for the outcome of the job. Something along the lines of `I run this job X and I expect it to do Y`. In this case, Y is that a message_bus message is published.